debezium-connector-cassandra/src/main/java/io/debezium/connector/cassandra/SourceInfo.java

```diff
@@ -108,4 +84,19 @@ public String toString() {
         map.put(TIMESTAMP_KEY, tsMicro);
         return map.toString();
     }
+
+    @Override
+    protected Instant timestamp() {
+        return Instant.ofEpochSecond(tsMicro/1000);
```

Comment:
See above, also should be converted keeping the precision.

Reply:
> See above, also should be converted keeping the precision.

Nice catch! Updated to use `Instant.ofEpochMilli(tsMicro)` to keep the precision.

Reply:
But `ofEpochMilli()` will loose precision, no?

Reply:
Oops, my bad. Updated to convert `micro-seconds` to `Instant`.

Reply:
> But `ofEpochMilli()` will loose precision, no?

Oops, my bad. Updated to convert `micro-seconds` to `Instant`.